### ISSUE TITLE
cannot .catch() on create

### ISSUE THREAD

**Issue by foxhound87** [feathersjs/feathers#359]
I cannot get the error from the catch() on create().
from the rest call is working and using same data I get:
E11000 duplicate key error index: aggregator.users.$email_1 dup key: { : "test@email.tld" }  

```
    return service('user')
      .create({ email, password, username })
      .catch((data) => console.log(data));
```

the error I get:

```
Error: [object Object]
    at Manager.eval (eval at <anonymous> (http://localhost:3100/bundle.js:2054:2), <anonymous>:21:15)
    at Decoder.eval (eval at <anonymous> (http://localhost:3100/bundle.js:2054:2), <anonymous>:21:15)
    at Socket.eval (eval at <anonymous> (http://localhost:3100/bundle.js:2054:2), <anonymous>:21:15)
From previous event:
    at Object.send (eval at <anonymous> (http://localhost:3100/bundle.js:10202:2), <anonymous>:49:14)
From previous event:
    at Object.mixin.(anonymous function) (eval at <anonymous> (http://localhost:3100/bundle.js:10190:2), <anonymous>:53:8)
    at Object.eval (eval at <anonymous> (http://localhost:3100/bundle.js:2030:2), <anonymous>:30:17)
    at Object.wrapper (eval at <anonymous> (http://localhost:3100/bundle.js:10274:2), <anonymous>:30:28)
    at Object.eval (eval at <anonymous> (http://localhost:3100/bundle.js:2030:2), <anonymous>:30:17)
    at Object.mixin.(anonymous function) (eval at <anonymous> (http://localhost:3100/bundle.js:10268:2), <anonymous>:17:32)
    at Object.eval [as create] (eval at <anonymous> (http://localhost:3100/bundle.js:2030:2), <anonymous>:30:17)
    at helper (eval at <anonymous> (http://localhost:3100/bundle.js:12398:2), <anonymous>:25:61)
    at helper (eval at <anonymous> (http://localhost:3100/bundle.js:12398:2), <anonymous>:42:12)
    at helper (eval at <anonymous> (http://localhost:3100/bundle.js:12398:2), <anonymous>:42:12)
    at access (eval at <anonymous> (http://localhost:3100/bundle.js:12398:2), <anonymous>:8:10)
    at dispatch (eval at <anonymous> (http://localhost:3100/bundle.js:1214:2), <anonymous>:51:37)
    at executeDispatchesAndRelease (eval at <anonymous> (http://localhost:3100/bundle.js:1148:2), <anonymous>:42:22)
    at executeDispatchesAndReleaseTopLevel (eval at <anonymous> (http://localhost:3100/bundle.js:1148:2), <anonymous>:53:10)
```

**Comment by UlyssesAlves** [comment]
Does it make any difference if you use it this way?

service('user') .create({ email, password, username }) .catch(data => {
    console.log(data)
});

**Comment by daffl** [comment]
What database adapter are you using and what does the client setup look like (REST, Sockets)? It might have something to do with https://github.com/feathersjs/feathers-errors/pull/43/files#diff-1fdf421c05c1140f6d71444ea2b27638R180 (the error has no `message` but is also not just a string) but to make sure we have to see what the original error looks like.

**Comment by ekryski** [comment]
I think @foxhound87 is using mongoose. If so, I have a PR ready that should fix this. https://github.com/feathersjs/feathers-mongoose/pull/92.

**Comment by ekryski** [comment]
@foxhound87 we've released feathers-mongoose@3.4.1. Let me know if that fixes it and we'll close this. Otherwise please post the actual error object. You'll probably need to log it out in the error handler.

**Comment by foxhound87** [comment]
@ekryski @daffl yes I'm using feathers-mongoose, now the error is show but with all the stack trace too.
There is no `message` object. Also, should be useful to have an user-friendly message for using in forms validation.

**Comment by foxhound87** [comment]
I don't know if this strange behavior of promises can be related to the same issue too: https://github.com/foxhound87/rfx-stack/issues/13

**Comment by daffl** [comment]
Can we close this? I'm not sure if it is fixed or what problem still remains. I think more user-friendly messages can be handled with afterError hooks (see https://github.com/feathersjs/feathers-hooks/issues/83)

**Comment by ekryski** [comment]
I think so. I fixed the original mongoose duplicate key error a little while ago in https://github.com/feathersjs/feathers-mongoose/pull/92.

I think the main issue is that if an error happens **at any time** you are inside the authenticate function it will throw an error and the `.catch()` on the `app.authenticate` call will catch it unless you have a `.catch()` call in your code elsewhere. This could be syntax errors, auth, connection errors, etc.

So I think we should just think of that catch like a global error handler when authenticating. Not sure if we can do anything else there. Closing.
